torchvision/datasets/utils.py

```diff
@@ -50,7 +51,55 @@ def bar_update(count, block_size, total_size):
     return bar_update
 
 
+_HASH_FUNCTIONS: Dict[int, _hashlib.HASH] = {
+    16: hashlib.md5(),
+    20: hashlib.sha1(),
+    28: hashlib.sha224(),
+    32: hashlib.sha256(),
+    48: hashlib.sha384(),
+    64: hashlib.sha512(),
+}
+
+
+def calculate_checksum(fpath: str, algorithm: _hashlib.HASH = hashlib.md5(), chunk_size: int = 1024 * 1024) -> str:
```

Comment:
(Looks like I forgot to submit this comment)

Do you need this function to be publicly exposed? IIUC you should only need `check_integrity`, so it would be best for us if we could keep `calculate_checksum` and `verify_checksum` private.

And in fact I would recommend bundling both in a single `_verify_checksum` function as we don't really need to extract out the logic of  `calculate_checksum`